packages/contracts/src/systems/FacilitySystem.sol

```diff
@@ -38,13 +47,35 @@ contract FacilitySystem is System {
   }
 
   /**
-   * @dev Check if a player owns a facility.
-   * @param player The address of the player to check.
-   * @param entityKey The key of the facility to check.
-   * @return True if the player owns the facility, false otherwise.
+   * @dev Check if a position is next to an existing entity.
+   * @param x The x coordinate of the position to check.
+   * @param y The y coordinate of the position to check.
+   * @param z The z coordinate of the position to check.
+   * @return True if the position is next to an existing entity, false otherwise.
+   */
+  function isPositionNextToAnExistingEntity(int32 x, int32 y, int32 z) public view returns (bool) {
+    if (!isPositionEmpty(x, y - 1, z)) return true;
+    if (!isPositionEmpty(x, y + 1, z)) return true;
+    if (!isPositionEmpty(x - 1, y, z)) return true;
+    if (!isPositionEmpty(x + 1, y, z)) return true;
+    if (!isPositionEmpty(x, y, z - 1)) return true;
+    if (!isPositionEmpty(x, y, z + 1)) return true;
+    return false;
+  }
+
+  /**
+   * @dev Check if a facility of entityTypeId can be built at the given position.
+   * @param entityTypeId The entityTypeId of the facility to check.
+   * @param x The x coordinate of the position to check.
+   * @param y The y coordinate of the position to check.
+   * @param z The z coordinate of the position to check.
+   * @return True if a facility of entityTypeId can be built at the given position, false otherwise.
    */
-  function isPlayerOwnerOfFacility(address player, bytes32 entityKey) public view returns (bool) {
-    return OwnedBy.get(entityKey) == player;
+  function canBuildFacilityTypeAtPosition(uint32 entityTypeId, int32 x, int32 y, int32 z) public view returns (bool) {
+    if (y < 0) return false;
+    if (!isPositionEmpty(x, y, z)) return false;
+    if (y == 0) return isEntityTypeIdGroundLevelFacility(entityTypeId);
+    return isPositionNextToAnExistingEntity(x, y, z);
```

Comment:
Yes nice, this makes sense. 
Observing that as we grow in scope we will have to considering how to keep client logic + contract logic in sync.